Reindexing fails in `prod` for non-disjunctive projects

The following datasets failed due to entities having invalid number of sources:

**CloudWatch Logs Insights**    
region: us-east-1    
log-group-names: /aws/lambda/azul-indexer-prod-aggregate    
start-time: 2025-10-18T10:36:47.766Z    
end-time: 2025-10-18T12:50:23.300Z    
query-string:
  ```
  fields @timestamp, @message
| filter @message like /ERROR/
| parse @message 'ERROR]*Traceback' as message
| parse message "*entity_id" as error
| parse message "name='*'" as dataset
| stats count(*) by dataset, error
| sort @timestamp desc
| limit 10000
  ```
---
| dataset | error | count(*) |
| --- | --- | --- |
| hca_prod_2079bb2e676e4bbf8c68f9c6459edcbb__20240327_dcp2_20250930_dcp54 | AssertionError: R(&#x27;Entity has an invalid number of sources&#x27;, CataloguedEntityReference(entity_type=&#x27;samples&#x27;, | 167 |
| hca_prod_4bcc16b57a4745bbb9c0be9d5336df2d__20240327_dcp2_20250930_dcp54 | AssertionError: R(&#x27;Entity has an invalid number of sources&#x27;, CataloguedEntityReference(entity_type=&#x27;samples&#x27;, | 111 |
| hca_prod_9c20a245f2c043ae82c92232ec6b594f__20220212_dcp2_20250930_dcp54 | AssertionError: R(&#x27;Entity has an invalid number of sources&#x27;, CataloguedEntityReference(entity_type=&#x27;samples&#x27;, | 87 |
---


All these errors had a similar Traceback:

```
Traceback (most recent call last):
  File "/var/task/azul/chalice.py", line 193, in patched_event_source_handler
    return old_handler(self_, event, context)
  File "/var/task/chalice/app.py", line 1761, in __call__
    return self.handler(event_obj)
  File "/var/task/azul/indexer/index_controller.py", line 168, in aggregate
    self.aggregate(event)
  File "/var/task/azul/indexer/index_controller.py", line 280, in aggregate
    self.index_queue_service.aggregate(tallies, retry=retry)
  File "/var/task/azul/indexer/index_queue_service.py", line 288, in aggregate
    self.index_service.aggregate(tally_by_entity)
  File "/var/task/azul/indexer/index_service.py", line 489, in aggregate
    assert len(aggregate.sources) == 1, R(
```

_Originally posted by @achave11-ucsc in https://github.com/DataBiosphere/azul/issues/7493#issuecomment-3423154858_

--- a/deployments/prod/environment.py
+++ b/deployments/prod/environment.py
@@ -1731,17 +1731,14 @@ dcp53_sources = mkdict(dcp52_sources, 518, mkdelta([
     mksrc('bigquery', 'datarepo-68e82765', 'hca_prod_f8aa201c4ff145a4890e840d63459ca2__20220119_dcp2_20250829_dcp53'),
 ]))
 
-dcp54_sources = mkdict(dcp53_sources, 524, mkdelta([
+dcp54_sources = mkdict(dcp53_sources, 521, mkdelta([
     mksrc('bigquery', 'datarepo-f1ae4f66', 'hca_prod_032880e52b444bfb8eef25bb48e7453f__20250930_dcp2_20250930_dcp54'),
     mksrc('bigquery', 'datarepo-ae2bf16b', 'hca_prod_16241d8231194bddbba55097c0591ba0__20250930_dcp2_20251001_dcp54'),
-    mksrc('bigquery', 'datarepo-17119788', 'hca_prod_2079bb2e676e4bbf8c68f9c6459edcbb__20240327_dcp2_20250930_dcp54'),
-    mksrc('bigquery', 'datarepo-261cfed5', 'hca_prod_4bcc16b57a4745bbb9c0be9d5336df2d__20240327_dcp2_20250930_dcp54'),
     mksrc('bigquery', 'datarepo-ed4ae538', 'hca_prod_5eafb94b02d8423e81b83673da319ca0__20220117_dcp2_20250930_dcp54'),
     mksrc('bigquery', 'datarepo-b06393f7', 'hca_prod_75dbbce90cde489c88a793e8f92914a3__20220117_dcp2_20250930_dcp54'),
     mksrc('bigquery', 'datarepo-3e921113', 'hca_prod_84d1697fd4af42c29a5037fb5842c586__20250203_dcp2_20250930_dcp54'),
     mksrc('bigquery', 'datarepo-fd40f5e4', 'hca_prod_87d52a86bdc7440cb84d170f7dc346d9__20220118_dcp2_20250930_dcp54'),
     mksrc('bigquery', 'datarepo-11fe0d2a', 'hca_prod_8dadf21e0a444fa3aafc7633b2a9eaa4__20250930_dcp2_20250930_dcp54'),
-    mksrc('bigquery', 'datarepo-be15ad3c', 'hca_prod_9c20a245f2c043ae82c92232ec6b594f__20220212_dcp2_20250930_dcp54'),
     mksrc('bigquery', 'datarepo-005330ae', 'hca_prod_9f7aa40170e34695951a30541a1434eb__20250203_dcp2_20250930_dcp54'),
     mksrc('bigquery', 'datarepo-29e38f9e', 'hca_prod_a004b1501c364af69bbd070c06dbc17d__20220118_dcp2_20250930_dcp54'),
     mksrc('bigquery', 'datarepo-6d20a90c', 'hca_prod_b32a9915c81b4cbcaf533a66b5da3c9a__20220118_dcp2_20250930_dcp54'),

